### ISSUE TITLE
Better error message for scoped type error

### ISSUE THREAD

**Issue by Gouthamve** [golang/go#15575]
1. What version of Go are you using (`go version`)?
`go version go1.6.2 darwin/amd64`

2. What operating system and processor architecture are you using (`go env`)?
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
```

3. What did you do?
https://play.golang.org/p/pJBsVvubBw

4. What did you expect to see?
An error message that was clear

5. What did you see instead?
Spent over an hour figuring out why `type a` was not taking `type a`

I think there should be a better error message for this case, as debugging it was super hard.

**Comment by griesemer** [comment]
It's not a common mistake: local types are rare, and local types named like outer types are even more rare, so I'd say there's no urgency on this.

That said, if we just say something like "type T (defined in F)" we could provide a very accurate message. Shouldn't be hard to do. It's a bit of a balance between precision of error and overly verbose error message.

**Comment by mdempsky** [comment]
It's also possible (albeit increasingly unlikely) for multiple type Ts to be declared within a single function, so there may be cases where we want to disambiguate identically named types by line number instead: https://play.golang.org/p/zs6_-3awLP

**Comment by griesemer** [comment]
@mdempsky Indeed. It probably would be better to just report the line number (or file:line info). Needs a bit of experimentation for a pleasant result.

**Comment by mdempsky** [comment]
CL 42750 adds support for line number disambiguation when using two different-but-identical-looking named types, but there are also cases where we have different-but-identical-looking anonymous types. For example:

    func _() {
        var s []int
        type int uint
        var t []int = s
        _ = t
    }

currently emits:

    cannot use s (type []int) as type []int in assignment

And in general, there might be arbitrarily many different named types within the source and destination types (e.g., maps and structs)

Are we happy to only special case named types? Or any ideas how to explain to the user why "map[K]V" and "map[K]V" might be different types?

**Comment by tzneal** [comment]
I didn't think about that.  It seems less likely to occur, but I don't like the inconsistency.  Happy to abandon the CL.

**Comment by griesemer** [comment]
Not urgent. Moving to 1.10.
